unstructured_ingest/v2/processes/connectors/gitlab.py

```diff
@@ -0,0 +1,249 @@
+        return False
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        project = self.connection_config.get_project()
+        ref = self.connection_config.git_branch or project.default_branch
+        git_tree = project.repository_tree(
+            ref=ref,
+            recursive=True,
+            iterator=True,
+            all=True,
+        )
+        for element in git_tree:
+            rel_path = element["path"].replace(self.connection_config.repo_path, "").lstrip("/")
+            if (
+                element["type"] == "blob"
+                and self.is_file_type_supported(element["path"])
+                and (
+                    not self.connection_config.git_file_glob
+                    or self.does_path_match_glob(element["path"])
+                )
+            ):
+                record_locator = {
+                    "file_path": element["path"],
+                    "file_name": element["name"],
+                    "file_type": element["type"],
+                    "repo_path": self.connection_config.repo_path,
+                }
+                if self.connection_config.git_branch is not None:
+                    record_locator["branch"] = self.connection_config.git_branch
+
+                yield FileData(
+                    identifier=element["id"],
+                    connector_type=CONNECTOR_TYPE,
+                    source_identifiers=SourceIdentifiers(
+                        fullpath=element["path"],
+                        filename=element["path"].split("/")[-1],
+                        rel_path=rel_path,
+                    ),
+                    metadata=FileDataSourceMetadata(
+                        url=element["id"],
+                        record_locator=record_locator,
+                        permissions_data=[{"mode": element["mode"]}],
+                    ),
+                    additional_metadata={},
+                )
+
+
+class GitLabDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class GitLabDownloader(Downloader):
+    connection_config: GitLabConnectionConfig
+    download_config: GitLabDownloaderConfig
+
+    @SourceConnectionNetworkError.wrap
+    @requires_dependencies(["gitlab"], extras="gitlab")
+    def _fetch_content(self):
+        from gitlab.exceptions import GitlabHttpError
+
+        try:
+            project = self.connection_config.get_project()
+            content_file = project.files.get(
+                self.path,
+                ref=self.connection_config.git_branch or project.default_branch,
+            )
+        except GitlabHttpError as e:
+            if e.response_code == 404:
+                logger.error(f"File doesn't exists {self.connection_config.url}/{self.path}")
+                return None
+            raise
+        return content_file
+
+    def _fetch_and_write(self, path) -> None:
+        content_file = self._fetch_content()
+        self.update_source_metadata(content_file=content_file)
```

Comment:
Where is this defined and what is it doing? I don't see an `update_source_metadata` method defined. It's something that's not typically used in the v2 approach.